### DISCUSSION TITLE
login problem

### DISCUSSION THREAD

**Discussion by junhou-he** [dani-garcia/vaultwarden#1866] [Q&A]
### Subject of the issue

The user accepted the email invitation and an error occurred while creating the account 
### Deployment environment

 

* vaultwarden version: 1.22.1

* Install method: Docker image

* Clients used: web vault

* Reverse proxy and version: nginx-1.20.1-2
* 

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:  
[error][ERROR] Error decoding JWT.
[CAUSE] Error(
    InvalidSignature,
)

### Steps to reproduce

docker run -d --name vaultwarden -v /data/:/data/ -p 8080:80 vaultwarden/server:latest
### Expected behaviour

The user accepts the email invitation and creates an account normally 
### Actual behaviour

The user accepted the email invitation and an error occurred while creating the account 
### Troubleshooting data

![Inked微信图片_20210720093133_LI](https://user-images.githubusercontent.com/76465560/126248698-c5080138-85bd-452e-983b-0b2da96d0454.jpg)
[logs.txt](https://github.com/dani-garcia/vaultwarden/files/6844884/logs.txt)

**Comment by BlackDex** [comment]
Either the user didn't accepted it soon enough, the token is only valid for 5 days. Or your security keys used to encode and decode those tokens were changed.

**Comment by junhou-he** [comment]
Hi BlackDex,

The user creates an account within five minutes of sending the invitation and reports the error. I didn't change any security keys.

**Comment by BlackDex** [comment]
Could you please generate a support string via`/admin/diagnostics`?

**Comment by junhou-he** [comment]
Hi ，

How to generate a support string, I can not find the path /admin/diagnostics

**Comment by BlackDex** [comment]
you need to go to `https://your.domain.tld/admin/diagnostics` that's it.

**Comment by junhou-he** [comment]
Hi,
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.22.1
* Web-vault version: v2.20.4b
* Running within Docker: true
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.35.4
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 


```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*****.*******-********.***",
  "domain_origin": "*****://*****.*******-********.***",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": true,
  "signups_allowed": true,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": false,
  "smtp_from": "*****@*******-********.***",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "*******.*******-********.***",
  "smtp_password": "***",
  "smtp_port": 25,
  "smtp_ssl": false,
  "smtp_timeout": 15,
  "smtp_username": "*****@*******-********.***",
  "templates_folder": "data/templates",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

**Comment by BlackDex** [comment]
Well i have tested this my self the following way:

```shell
# Terminal 1
docker run --rm --name vaultwarden \
  -p 8080:80 \
  -e DOMAIN=http://127.0.0.1:8080 \
  -e SMTP_HOST=172.17.0.1 \
  -e SMTP_PORT=1025 \
  -e SMTP_SSL=false \
  -e SMTP_FROM=vw@vw.local \
  vaultwarden/server:latest

# Terminal 2
docker run --rm --name maildev \
  -p 1080:1080 \
  -p 1025:1025 \
  maildev/maildev
```

Within a browser i opened `http://127.0.0.1:1080` and `http://127.0.0.1:8080`.
Then i created an account, and organization, and after that I invited a user. Checked the MailDev interface and used the received link to create an account. All works without any issues.

I also tested `vaultwarden/server:testing` same result.

**Comment by junhou-he** [comment]
Okay ,thank you. The error may be caused by improper operation by the user, I will close this issue
